### PR TITLE
Fix remove ring

### PR DESCRIPTION
**Pull request by Xynnn007**
Fixes: #134

### CODE REVIEW

**Comment by Xynnn007** [comment]
Night error issue https://github.com/confidential-containers/image-rs/issues/136

**Comment by Xynnn007** [comment]
The errors of `nightly` and `beta` is tracked by https://github.com/confidential-containers/image-rs/issues/136

Hi @jiangliu I've changed previous CI tests for different combinations of features. Please take a look if it looks good to you.

Hi @stevenhorsman  @BbolroC This PR will help `s390x` to support cosign. A question is that

File: Cargo.toml [hunk 1]
```diff
@@ -76,12 +73,23 @@ members = ["libs/test-utils"]
...
+# This will be based on `ring` dependency
+kata-cc-rustls = ["encryption-ring", "keywrap-ttrpc", "snapshot-overlayfs", "signature-cosign-rustls", "signature-simple", "getresource", "rustls-tls"]
+enclave-cc-rustls = ["encryption-ring", "keywrap-native", "eaa-kbc", "snapshot-unionfs", "signature-simple", "getresource", "signature-cosign-rustls", "rustls-tls"]
+
+# This will be based on `openssl` dependency
+kata-cc-native = ["encryption-openssl", "keywrap-ttrpc", "snapshot-overlayfs", "signature-cosign-native", "signature-simple", "getresource", "native-tls"]
+enclave-cc-native = ["encryption-openssl", "keywrap-native", "eaa-kbc", "snapshot-unionfs", "signature-simple", "getresource", "signature-cosign-native", "native-tls"]
+
+native-tls = ["oci-distribution/native-tls"]
+rustls-tls = ["oci-distribution/rustls-tls"]
```
Comment (arronwy) [inline Cargo.toml:87]: can we rename `native-tls` and `rustls-tls` with prefix since these two features seems global and may confusing.

**Review by arronwy — APPROVED** [review]
Thanks @Xynnn007 , LGTM！

**Comment by stevenhorsman** [comment]
@Xynnn007 - just to double check - this mean we can remove the separate `s390x` dependency pulled in in the `kata-agent` now? 
```
# Image pull/decrypt
[target.'cfg(target_arch = "s390x")'.dependencies]
image-rs = { git = "https://github.com/confidential-containers/image-rs", tag = "v0.5.1", default-features = false, features = ["kata-cc-s390x"] }

[target.'cfg(not(target_arch = "s390x"))'.dependencies]
image-rs = { git = "https://github.com/confidential-containers/image-rs", tag = "v0.5.1", default-features = false, features = ["kata-cc"] }
```
If so I'll try removing it and enabling the co-sign tests for s390x (we might have to mutli-arch enable/create the s390x versions of those to get them passing, but it will be good to understand and work on that early in the release cycle.

Thanks for the great work!

**Pull request merged**